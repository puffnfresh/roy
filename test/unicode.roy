let named x = x
let anon = \x -> x

let multiArgs x y z = x
let anonMultiArgs = \x y z -> x

console.log (named "Hello") (anon "world")
console.log (multiArgs "Hello" "world" "!") (anonMultiArgs "world" "!" "Hello")


let traceMonad = {
  return: \x ->
    console.log "Return:" x
    x
  bind: \x f ->
    console.log "Binding:" x
    f x
}

console.log (do traceMonad
  w <- 1
  let x = 2
  y <- 3
  z <- 4
  return w + x + y + z
)

let тест = "test"
console.log тест
