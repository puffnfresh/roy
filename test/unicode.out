Hello world
Hello world
Binding: 1
Binding: 3
Binding: 4
Return: 10
10
test
